### PR TITLE
Add "After The Fall" era featuring Dunefolk and Quenoth while removing Dunefolk from Default Era

### PR DESCRIPTION
**Pull request by Jonathan-Kelly**
Also tentatively adds Drakes to the new After The Fall era to fill the role of a third faction (since they also do well on sand heavy maps).

### Create a place for new game mechanics and balance goals to be explored:
* real ranged attacks
* area-of-affect attacks
* finer terrain movement cost system (as [suggested](https://github.com/wesnoth/wesnoth/pull/8688#issuecomment-2059383549) by @soliton-)
* better mirror match ups
* more abilities and specials per unit
* concepts from [Project Haldric](https://github.com/Byteron/haldric-next)

### More room for additional factions after Dunefolk.

Each new faction added to an era greatly increases balance complexity, so if adding a 7th faction to Default Era is already highly questionable then adding an 8th would be rather insane. At the same time, you need a minimum of 2 or 3 factions to get a new era started. Using Dunefolk and Quenoth to start a new era opens room for 3 or 4 addition factions to be added over the next decade of Wesnoth development. Possibilities include a version of the Clockwork Dwarves or one based on the Ixthala demons as replacements for the Undead (as suggested by dashomew/matto).

### Neatly fit Dunefolk into the Lore in a meaningful way.

The Dunefolk's fear of magic is clearly justified if they are the human survivors of The Fall, with their world having been ruined through the use of magic, while tyrannical magical entities dominate what is left. Other After The Fall factions can be made more fantastical and magical to better contrast with and highlight the thematic mundane-ness of the Dunefolk. Whereas Default era already has many non-magical units and even whole factions like the Knalgan Alliance and Northerners, so adding the Dunefolk to that time period does little for them or it.

Having an After The Fall era helps fill out the UtBS time period and make UtBS feel more "canonical" compared with the rest of the mainline campaigns. Units, abilities and specials made for After The Fall era can be used in the creation of additional campaigns set in UtBS's time period, too.

### Avoid imbalancing the venerable and tightly balanced Default era.

Default era has been stable and nearly unchanged from 1.8 (released in 2010) to 1.14/1.16 after having been put through numerous tournaments back when the competitive community was large and thriving. Wheras the Dunefolk faction is very poorly tested in highly competitive environments, having been rarely if ever used in tournaments and not adopted by the competitive community. It is likely poorly balanced compared to the classic six factions.

### Avoid needlessly antagonizing fans of classic Default era.

We don't need to try to force/trick fans of the classic 6 faction Default era (and its derived balance mods used in the tournaments) to play an experimental faction in order to advance mainline multiplayer. A new era allows for fresh multiplayer content to be created without further upsetting traditional players as #8688 is otherwise likely to do when 1.20 is released.

The highly unpopular 1.18 and (proposed) 1.20 changes to Default era units have already caused enough controversy, distrust and division. We should be moving away from this kind of approach.

## Art Limits

One thing we might strive for with this era is more judicious use of unit sprites, since they are the biggest cost of a multiplayer era. Some ideas:
* fewer units per faction (4-6 rather than 6-8)
* use existing mainline sprites (scorpion, jinn, fire guardian)
* use sprites from user made content

### CODE REVIEW

**Comment by Pentarctagon** [comment]
Reusing the Drakes seems very odd. Are there really no other factions that could be used?

**Comment by Jonathan-Kelly** [comment]
Drakes are the best option among the factions which are _already_ in mainline. There's countless options out on the add-ons server which could meet the thematic requirements for ATF era, including even Steelhive if we consider the ending of UtBS. But trying to decide on additional factions here and now would likely completely bog down this PR and make its comments section reach up to the high hundreds.

Likely none of this era's factions or their units' stats will stay in their current state after this PR. Those will change greatly depending on what game mechanics we want to explore/use in this era. Though the sprites and other art assets are more likely to see reuse.

**Comment by Pentarctagon** [comment]
Despite the wording in the PR, the Dunefolk getting added to the Default era is not some catastrophic thing that requires adding an After the Fall era to the game in such an incomplete state.

**Comment by Gothyoba** [comment]
Pleasee keep in mind that an AF Era to be proposed in in development, and nobody so far has proposed drakes to be in it. The drakes suggestion is honestly extremely weird.

**Comment by Jonathan-Kelly** [comment]
> adding an After the Fall era to the game in such an incomplete state.

In its current state the era is working and playable on the technical side of things. But it is reaching the point where it could use some wider feedback about what creative direction we would like to take it in. Particularly regarding which new game mechanics to apply to it, since that will influence a lot of other things.

> Despite the wording in the PR, the Dunefolk getting added to the Default era is not some catastrophic thing

I apologize for my wording; it has been a long week and I was feeling it when I wrote the summary.

The one specific point I wanted to address there was not about Dunefolk getting added to Default era but the simultaneous removal of the classic 6 faction era after 20 years (instead of just renaming it to [Classic Era](https://github.com/wesnoth/wesnoth/pull/8688#issuecomment-2037743791) or something to that effect). Your [concern](https://github.com/wesnoth/wesnoth/pull/8688#issuecomment-2037929772) that most people might be playing the 6 faction default era simply because it loads by default is reasonable by itself. But as I said [before](https://github.com/wesnoth/wesnoth/pull/8688#issuecomment-2039959339):

_If your thinking is correct that the name "Default" and its default-ness are the only reasons for its considerable popularity, then this will be confirmed by the 1.20 player statistics showing that the new default era maintains the same relative player count as the old one did in 1.18, while a 1.20 "Classic" era drops down to the same usage level as Age of Heroes. But otherwise, if a large chunk of players still prefers Classic, then they will not be disappointed by seeing it taken away._

In other words, allow players to have the same era choices available in 1.20 as they do in 1.18 and let the statistical evidence speak for itself about what they like. If few players actively choose to play the classic 6 faction era then we could safely drop it from 1.22.

**Comment by Pentarctagon** [comment]
I think that this needs at minimum one new faction added to it. I also don't think the Drakes should be added since I can only imagine the kind of drama that needing to keep them in sync between both eras will cause. And no, I'm not going to engage in restarting the debate from #8688.

**Comment by knyghtmare** [comment]
I have a lot of things to state here:

- **This should be an addon first**. Adding it directly to the game core might come off as developer malpractice (wouldn't be the first time a dev did that). Also, doing this will only incite backlash and you will be unable to push out updates until the same monthly version release. You sure this is a wise decision? I dont think it is. At all.

- > There's countless options out on the add-ons server which could meet the thematic requirements for ATF era, including even Steelhive if we consider the ending of UtBS.

   Factions like Steelhive and others heavily used in a lot of UMC. Adding these to core in "ATF-tailored" variants might break several in terms of balance advancement paths as core unit IDs cannot be overridden as image paths can. If you do have to take sprites for new units, **I suggest you rebrand the ID prefixes, such as Metalhive or sth to ensure UMC stays unaffected**. Leave the existing id prefixes alone. **Completely reevaluation of stats of factions is equivalent to making a new faction so no need to invade UMC area with unit type ID theft.**

- I have been monitoring a discussion channel at the WMG server about the development of ATF Era and what they have theory crafted so far, seems quite more interesting, the same foundation (DF and Quenoth Elves plus 4 other factions, including 1 being made from base zero). You are more than welcome to join in and make it more collaborative than whatever it is you're trying to do individually. (I see 5 thumbed down).

**Comment by ForestDragon-wesnoth** [comment]
I agree with knyght that we shouldn't reuse ids from popular UMC content, and that such a project would be best done as an add-on first to reduce controversy of it getting added

Telling the UMC authors to change ID prefixes, when several UMC already use the faction of that era as dependency seems disrespectful and would be yet another instance of developer malpractice.

**Comment by soliton-** [comment]
Confusing what issues of ID theft has to do with this PR. I guess there is context elsewhere that makes these comments make sense...

---

This PR points in the right direction but I agree that development as an addon would be preferable. Adding default era factions like drakes would only make sense for play testing if the goal is to establish a similar balance to default but IMO the "opposite" should be the goal. A new era should play significantly differently to default era. Which also includes no expectation that existing maps for default era matches produce balanced match ups using the new era.

**Comment by Pentarctagon** [comment]
I also think the expectation should be set for this new era that it's not something that will necessarily be essentially permanently stable once 1.20 is out. It shouldn't be a museum piece like Default era basically is, it should be something that's allowed to change more frequently and significantly to add interesting gameplay features, shake up the meta, etc.

**Comment by soliton-** [comment]
Default era simply already went through a balancing phase with significant changes years ago so all that is left is minor adjustments. I agree that the expectation for a new era should be that it will undergo significant changes early on. (So unlikely to be that stable by the time of 1.20) And it would be weird if that somehow goes on indefinitely instead of stabilizing over time like default era.

I don't think "shaking up the meta" is a sensible goal for wesnoth if it means changing the balance for the sake of it. Something like that _may_ be viable if there is a huge player base that actually explores the "meta" sufficiently to get bored of it and there are enough developers available to shift the balance around and adjust every affected scenario accordingly. Wesnoth is quite far from that though so it feels more like creating busy work for developers and players alike for no benefit. (I'd rather expect it would drive people away.)

**Comment by MechanicalDragon963** [comment]
Alright, I joined Jonathan Kelly and we started development on After the Fall Era as an addon.

**Comment by Pentarctagon** [comment]
Closing due to age - feel free to re-open though.
